python/pyspark/sql/dataframe.py

```diff
@@ -2045,7 +2045,46 @@ def toDF(self, *cols):
         """
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
-
+    
+    @since(3.0)
+    def transform(self, func, *args, **kwargs):
```

Comment:
I think we don't need args and kwargs to be consistent with Scala side.

Reply:
Yea but let's keep this consistent with Scala side for now. Other Python APIs are historically consistent with Scala side. Let's add other variants later when it's absolutely required.